File: src/test/java/utils/codec/Base64Test.java
package utils.codec;

import org.junit.Test;
import static org.junit.Assert.*;

public class Base64Test {
    @Test
    public void toBase64Test() {
        String result1 = Base64.toBase64(new byte[] { 77, 97, 110 });
        String result2 = Base64.toBase64(new byte[] { 77, 97 });
        String result3 = Base64.toBase64(new byte[] { 77 });

        assertEquals("TWFu", result1);
        assertEquals("TWE=", result2);
        assertEquals("TQ==", result3);
    }

    @Test
    public void fromBase64Test() {
        byte[] result1 = Base64.fromBase64("TWFu");
        byte[] result2 = Base64.fromBase64("TWE=");
        byte[] result3 = Base64.fromBase64("TQ==");

        assertArrayEquals(new byte[] { 77, 97, 110 }, result1);
        assertArrayEquals(new byte[] { 77, 97 }, result2);
        assertArrayEquals(new byte[] { 77 }, result3);
    }
}

File: src/test/kotlin/utils/codec/Base64Test.kt
package utils.codec

import io.kotest.core.spec.style.FunSpec
import io.kotest.datatest.withData
import io.kotest.matchers.shouldBe
import io.kotest.property.Arb
import io.kotest.property.arbitrary.int
import io.kotest.property.arbitrary.uByte
import io.kotest.property.arbitrary.uByteArray
import io.kotest.property.checkAll


@ExperimentalUnsignedTypes
class Base64Test : FunSpec({
    val testCases = listOf(
        Pair(Base64("TWFu"), ubyteArrayOf(77u, 97u, 110u)),
        Pair(Base64("TWE="), ubyteArrayOf(77u, 97u)),
        Pair(Base64("TQ=="), ubyteArrayOf(77u))
    )

    context("Given test cases pass") {
        withData(testCases) { (base64, bytes) ->
            val uBytes = bytes.toUByteArray()
            base64.toBytes() shouldBe uBytes
            uBytes.toBase64() shouldBe base64
        }
    }

    context("Property test") {
        test("From bytes to base64 to bytes") {
            checkAll(Arb.uByteArray(Arb.int(1..20), Arb.uByte())) {
                it.toBase64().toBytes() shouldBe it
            }
        }
    }
})
